validation/views.py

```diff
@@ -707,6 +734,59 @@ def save_wrong_word(request, recording_id):
     return response
 
 
+@login_required()
+def record_audio(request):
+    if request.method == "POST":
+        form = RecordNewPhrase(request.POST, request.FILES)
+        translation = request.POST.get("translation")
+        transcription = request.POST.get("transcription")
+        audio_data = request.FILES["audio_data"]
+
+        phrase = Phrase.objects.filter(transcription=transcription).first()
+        if not phrase:
+            phrase = Phrase(
+                translation=translation,
+                field_transcription=transcription,
+                transcription=transcription,
+                kind=Phrase.SENTENCE if " " in transcription else Phrase.WORD,
+                status=Phrase.USER,
+                date=datetime.datetime.now(),
+                modifier=request.user,
+            )
+            phrase.save()
+
+        speaker = Speaker.objects.filter(user=request.user).first()
+        if not speaker:
+            speaker = Speaker(
+                full_name=request.user.first_name + " " + request.user.last_name,
+                code=request.user.first_name[:3].upper(),
```

Comment:
The codes are already **super**-ambiguous and confusing. Automatically adding more is inviting disaster. What will happen if someone else named Brian signs up??

Could we use the user’s login id, e.g., `andrew`, or `U${user.id}` like `U15` for user-submitted? That’ll at least keep it at three characters for the first 100 people.

Reply:
I know we have these codes from importing data that originally was partly on paper. I know we need to keep these codes around to link to that. But I’d recommend a general change to speech-db where, if there’s a user object for any speaker, show “$username” in the interface instead of this speaker code.

If we’re gathering data with a computer into a relational database in the first place, we don’t need to keep using concepts invented as poor workarounds for not having a database, especially concepts that are both hard-to-understand and ambiguous.

Reply:
This is something I was worried about before, yeah. The username is guaranteed to be unique, so I can use that as the speaker code for now at least.